lib/pf/firewallsso/Checkpoint.pm

```diff
@@ -0,0 +1,110 @@
+package pf::firewallsso::Checkpoint;
+
+=head1 NAME
+
+pf::firewallsso::Checkpoint
+
+=head1 SYNOPSIS
+
+The pf::firewallsso::Checkpoint module implements an object oriented interface
+to update the Checkpoint user table.
+
+=cut
+
+
+use strict;
+use warnings;
+
+use base ('pf::firewallsso');
+
+use POSIX;
+use Log::Log4perl;
+
+use pf::config;
+sub description { 'Checkpoint Firewall' }
+use pf::util::radius qw(perform_rsso);
+use pf::node qw(node_view);
+use pf::accounting qw(node_accounting_current_sessionid);
+
+=head1 METHODS
+
+=head2 action
+
+Perform a radius accounting request based on the registered status of the node and his role.
+
+=cut
+
+sub action {
+    my ($self,$firewall_conf,$method,$mac,$ip,$timeout) = @_;
+    my $logger = Log::Log4perl::get_logger(ref($self));
+
+    my $node_info = node_view($mac);
+
+    my @categories = split(/,/, $ConfigFirewallSSO{$firewall_conf}->{categories});
+    if (
+        defined($node_info) &&
+        (ref($node_info) eq 'HASH') &&
+        $node_info->{'status'} eq $pf::node::STATUS_REGISTERED &&
+        (grep $_ eq $node_info->{'category'}, @categories)
+    ){
+        my $username = $node_info->{'pid'};
+        $username = $node_info->{'last_dot1x_username'} if ( $ConfigFirewallSSO{$firewall_conf}->{'uid'} eq '802.1x');
+        return 0 if ( $ConfigFirewallSSO{$firewall_conf}->{'uid'} eq '802.1x' && $node_info->{'last_dot1x_username'} eq '');
+        my $acctsessionid = node_accounting_current_sessionid($mac);
+        my $connection_info = {
+          nas_ip => $firewall_conf,
+          nas_port => $ConfigFirewallSSO{$firewall_conf}->{'port'},
+          secret => $ConfigFirewallSSO{$firewall_conf}->{'password'},
+        };
+
+        my $attributes = {
+            'Acct-Session-Id' =>  $acctsessionid,
+            'Acct-Status-Type' => $method,
+            'User-Name' => $username,
+            'Session-Timeout' => $timeout,
+            'Called-Station-Id' => '00:11:22:33:44:55',
+            'Framed-IP-Address' => $ip,
+            'Calling-Station-Id' => $mac,
+        };
+        my $vsa = [];
+
+        perform_rsso($connection_info,$attributes,$vsa);
+        $logger->info("Node $mac registered and allowed to pass the Checkpoint Firewall");
+        return 1;
+    } else {
+        return 0;
```

Comment:
We try to avoid 1 or 0 in the case of TRUE or FALSE... or whatnot... there's sure exist a constant that will fit your need.